### PR TITLE
Add IsDigraphCore property

### PR DESCRIPTION
**Pull request by MTWhyte**
This pull request adds a method for determining whether or not a digraph is a core. A digraph _D_ is a core if and only if every endomorphism on _D_ is an automorphism on _D_.

### CODE REVIEW

File: gap/prop.gi [hunk 1]
```diff
@@ -589,3 +589,48 @@ end);
...
+                             hook,                      # hook
+                             fail,                      # user_param
+                             infinity,                  # max results
+                             fail,                      # hint
+                             false,                     # injective
+                             DigraphVertices(D),        # image
+                             [],                        # partial_map
+                             fail,                      # colors1
+                             fail);                     # colors2
+  if not proper_endo_found then
```
Comment (wilfwilson) [inline gap/prop.gi]: These five lines can be replaced with
`return not proper_endo_found;`.

File: gap/prop.gi [hunk 2]
```diff
@@ -589,3 +589,48 @@ end);
...
+  N := DigraphNrVertices(D);
+  if DigraphHasLoops(D) and N > 1 then
+    return false;
+  elif IsCompleteDigraph(D) then
+    return true;
+  elif IsBipartiteDigraph(D) and N > 2 then
+    return false;
+  fi;
+  # The core of a digraph with loops is a vertex with a loop, and the core of a
+  # bipartite digraph is the complete digraph on 2 vertices.
```
Comment (wilfwilson) [inline gap/prop.gi]: A non-empty bipartite digraph. If there’s no edges the core is the empty digraph with one vertex.

**Comment by MTWhyte** [comment]
Thanks for your suggestions - I've implemented them now. Before I push again, I'm thinking about the empty digraph with no vertices. We have

~~~
gap> D := EmptyDigraph(0);
<immutable digraph with 0 vertices, 0 edges>
gap> DigraphHomomorphism(D, D);
fail
~~~

and so by the above `D` here shouldn't be a core. On the other hand, if `D` isn't a core, then what would its core be, given that it has no proper subdigraphs? What do you think?

**Comment by wilfwilson** [comment]
Good idea about the digraph with no vertices.

I think that the digraph with 0 vertices is a core. If you think of a function on a set as a special kind of binary relation on that set, then the unique function on the digraph with 0 vertices is simply the empty relation, i.e. the empty function, which vacuously is an automorphism. So everything is consistent :)

I think it's fair enough that `DigraphHomomorphism(D, D);` fails here, for technical reasons rather than mathematical reasons, because it's trick to implement an 'empty function' in GAP.

**Comment by MTWhyte** [comment]
Great, I'll push my changes now

**Pull request merged**